remove k3d dep and namespace the namespace

dviersel opened the pull request:
Nice to have the separate namespaces. It might be an idea to use a namespace variable as input to the templates. Now the "namespace logic" is spread out over multiple files/places.

stokpop committed:
Update Dockerfile, build scripts and versions in pom-init.xml

Updated the Docker buildx commands in our build.sh files for K6, Gatling, and JMeter to use the --no-cache option for cleaner builds. Use arch specific build for g2i.

dviersel committed:
Upgrade to Docker build script and Kubernetes scripts

This update comes with several changes to improve compatibility and performance. The Docker build script is revised to use rsync instead of wget for copying files, which should provide a more reliable transfer process. Moreover, the Kubernetes scripts are modified to remove the dependency on k3d.

dviersel committed:
Update PerfanaConfigGeneratorController and add setup instructions

Modified PerfanaConfigGeneratorController to change return type from FileUploadResponse to String so that the return value can be used by another script or program. Added SETUP.md in the template directory, providing detailed setup instructions for using the Perfana starter package. Also, a new bash script file, create-starter-package.sh, has been added in the scripts directory to create the starter package.

PeterPaul-Perfana commented on template/grafana-agent.yaml.hbs:
The url and targets also contain the namespace, need to be updated

dviersel replied:
fixed

PeterPaul-Perfana reviewed (commented):
See comments

dviersel committed:
Update Grafana URLs and apiKey in templates

Grafana URLs in the template/grafana-agent.yaml.hbs file have been updated to include the client variable. In the maven/docker-gatling/pom-init.xml file, the PERFANA_API_KEY is restored.

dviersel commented on maven/docker-gatling/pom-init.xml:
```diff
@@ -56,7 +56,13 @@
         <version>v1.0.0</version>
         <tags>gatling,spring-boot-kubernetes</tags>
-        <apiKey>${ENV.PERFANA_API_KEY}</apiKey>
+        <apiKey>export PERFANA_CONFIG_EXAMPLES_DIR=$(pwd)/perfana-config-examples
```
pasted a snippet by accident; is now reverted

dviersel committed:
move the script to aws project
